src/apps/explorer/components/common/Dropdown/index.tsx

```diff
@@ -0,0 +1,160 @@
+  PositionLeftCSS,
+  PositionRightCSS,
+  DirectionDownwardsCSS,
+  DirectionUpwardsCSS,
+  DropdownItemProps,
+  DropdownItemCSS,
+} from './styled'
+import useOnClickOutside from 'hooks/useOnClickOutside'
+
+export enum DropdownPosition {
+  center,
+  left,
+  right,
+}
+
+export enum DropdownDirection {
+  downwards = 'down',
+  upwards = 'up',
+}
+
+const Wrapper = styled.div<{ isOpen: boolean; disabled: boolean }>`
+  outline: none;
+  pointer-events: ${(props): string => (props.disabled ? 'none' : 'initial')};
+  position: relative;
+  z-index: ${(props): string => (props.isOpen ? '100' : '50')};
+
+  &[disabled] {
+    cursor: not-allowed;
+    opacity: 0.5;
+  }
+`
+const ButtonContainer = styled.div`
+  background-color: transparent;
+  border: none;
+  display: block;
+  outline: none;
+  padding: 0;
+  user-select: none;
+  width: 100%;
+`
+export interface DropdownProps extends DOMAttributes<HTMLDivElement> {
+  activeItemHighlight?: boolean | undefined
+  className?: string
+  closeOnClick?: boolean
+  currentItem?: number | undefined
+  disabled?: boolean
+  items: Array<ReactElement>
+  triggerClose?: boolean
+  dropdownButtonContent?: React.ReactNode | string
+  dropdownDirection?: DropdownDirection | undefined
+  dropdownPosition?: DropdownPosition | undefined
+}
+
+type CssString = FlattenSimpleInterpolation | string
+
+const Items = styled(BaseCard)<{
+  dropdownDirection?: DropdownDirection
+  dropdownPosition?: DropdownPosition
+  fullWidth?: boolean
+  isOpen: boolean
+}>`
+  background: ${({ theme }): string => theme.bg1};
+  border-radius: 0.6rem;
+  border: 1px solid ${({ theme }): string => theme.borderPrimary};
+  box-shadow: ${({ theme }): string => theme.boxShadow};
+  display: ${(props): string => (props.isOpen ? 'block' : 'none')};
+  min-width: 160px;
+  position: absolute;
+  white-space: nowrap;
+
+  ${(props): CssString => (props.fullWidth ? 'width: 100%;' : '')}
+  ${(props): CssString => (props.dropdownPosition === DropdownPosition.left ? PositionLeftCSS : '')}
+  ${(props): CssString => (props.dropdownPosition === DropdownPosition.right ? PositionRightCSS : '')}
+  ${(props): CssString => (props.dropdownPosition === DropdownPosition.center ? PositionCenterCSS : '')}
+  ${(props): CssString => (props.dropdownDirection === DropdownDirection.downwards ? DirectionDownwardsCSS : '')}
+  ${(props): CssString => (props.dropdownDirection === DropdownDirection.upwards ? DirectionUpwardsCSS : '')}
+`
+
+Items.defaultProps = {
+  dropdownDirection: DropdownDirection.downwards,
+  dropdownPosition: DropdownPosition.left,
+  fullWidth: false,
+  isOpen: false,
+}
+
+export const DropdownOption = styled.li<DropdownItemProps>`
+  ${DropdownItemCSS}
+  list-style-type: none;
+`
+
+export const Dropdown: React.FC<DropdownProps> = (props) => {
+  const {
+    activeItemHighlight = true,
+    className = '',
+    closeOnClick = true,
+    currentItem = 0,
+    disabled = false,
+    dropdownButtonContent = '▼',
```

Comment:
Probably up to the caller, but would be cool to invest the direction when dropdown is expanded.

Anyway, I'm sure @biocom will suggest a fancy css or png triangle for this instead of the unicode symbol

Reply:
@alfetopito
I have applied it.